src/test/java/tests/pages/DashboardPage.java

```diff
@@ -0,0 +1,121 @@
+package tests.pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+import static core.utils.Timeout.TWO_SEC;
+import static org.junit.Assert.assertTrue;
+
+public class DashboardPage extends PageFactory {
+  public static final String TITLE_ALL_DASHBOARDS_XPATH = "//span[@title = 'All Dashboards']";
+  public static final String DEMO_DASHBOARD_LINE_XPATH = "//div[contains(@class, 'gridRow')]//a[text() = 'DEMO DASHBOARD']";
+  public static final String WIDGETS_GRID_XPATH = "//div[contains(@class, 'react-grid-item widgetsGrid')]";
+  public static final String RESIZE_WIDGETS_BUTTON_XPATH = "//span[@class ='react-resizable-handle react-resizable-handle-se']";
+  public static final String WIDGET_XPATH_TEMPLATE = "//div[contains(@class, 'react-grid-item widgetsGrid') and contains(., '%s')]";
+  private final WebDriver driver;
+
+  public DashboardPage(WebDriver driver) {
```

Comment:
I'd propose to create base page class with constructor:
```
  public BasePage(WebDriver driver) {
    this.driver = driver;
    initElements(driver, this);
  }
```
and in all parent page classes just use:
```
  public DashboardPage(WebDriver driver) {
    super(driver)
  }
```